it/src/test/java/com/linecorp/centraldogma/it/mirror/git/ForceRefUpdateTest.java

```diff
@@ -0,0 +1,228 @@
+
+    @RegisterExtension
+    static GitFolderExtension git = new GitFolderExtension() {
+        @Override
+        protected boolean runForEachTest() {
+            return true;
+        }
+    };
+
+    @RegisterExtension
+    static final CentralDogmaExtension dogma = new CentralDogmaExtension() {
+        @Override
+        protected void configure(CentralDogmaBuilder builder) {
+            builder.mirroringEnabled(true);
+        }
+    };
+
+    private static MirroringService mirroringService;
+    private String projName;
+    private KeyPair keyPair;
+    private String privateKey;
+    private String publicKey;
+
+    @BeforeAll
+    static void init() {
+        mirroringService = dogma.mirroringService();
+    }
+
+    @BeforeEach
+    void initDogmaRepo(TestInfo testInfo) throws NoSuchAlgorithmException {
+        projName = TestUtil.normalizedDisplayName(testInfo);
+        dogma.client().createProject(projName).join();
+        dogma.client().createRepository(projName, REPO_FOO).join();
+
+        final KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+        keyPair = kpg.generateKeyPair();
+        privateKey = Jackson.escapeText(KeyPairUtilsTest.toPemFormat(keyPair.getPrivate()));
+        publicKey = Jackson.escapeText(KeyPairUtilsTest.toPemFormat(keyPair.getPublic()));
+    }
+
+    @AfterEach
+    void afterEach() {
+        dogma.client()
+             .forRepo(projName, Project.REPO_META)
+             .commit("cleanup",
+                     Change.ofRemoval("/credentials.json"),
+                     Change.ofRemoval("/mirrors.json"))
+             .push().join();
+    }
+
+    @Test
+    void testLocalToRemote() throws Exception {
+        final AtomicBoolean throttleGitPush = new AtomicBoolean();
+
+        try (SshServer sshd = SshServer.setUpDefaultServer()) {
+            final GitLocationResolver resolver = GitLocationResolver.constantPath(git.gitWorkTree().toPath());
+            sshd.setCommandFactory(new GitPackCommandFactory(resolver) {
+                @Override
+                public GitPackCommand createGitCommand(String command) {
+                    if (command.contains("git-receive-pack") && throttleGitPush.get()) {
+                        throw new RuntimeException();
+                    }
+                    return super.createGitCommand(command);
+                }
+            });
+            final KeySetPublickeyAuthenticator authenticator =
+                    new KeySetPublickeyAuthenticator("keypair-auth", ImmutableList.of(keyPair.getPublic()));
+            sshd.setPublickeyAuthenticator(authenticator);
+            sshd.setKeyPairProvider(KeyPairProvider.wrap(keyPair));
+            sshd.start();
+
+            final String gitUri = "git+ssh://127.0.0.1:" + sshd.getPort() + "/.git/";
+            pushMirror(gitUri, MirrorDirection.LOCAL_TO_REMOTE);
+            pushCredentials(publicKey, privateKey);
+
+            // 1. initial mirroring
```

Comment:
Perform the initial mirroring?
```suggestion
            // 1. Perform the initial mirroring.
```